pkg/controller/controller.go

```diff
@@ -326,18 +326,25 @@ func (c *Controller) initSharedInformers() {
 func (c *Controller) Run(stopCh <-chan struct{}, wg *sync.WaitGroup) {
 	c.initController()
 
+	// start workers reading from the events queue to prevent the initial sync from blocking on it.
```

Comment:
that basically means that workers must always start before `acquireInitialListOfClusters`  executes since it queues the 1st Sync event, right ?

Reply:
Yes, and if there are no consumers of that queue `acquireInitialListOfClusters`  function will block forever...